blogs/2025/07/15/copilot-coding-agent.md

```diff
@@ -0,0 +1,89 @@
+
+## Copilot Coding Agent in Visual Studio Code
+
+Copilot Coding Agent integrates with the [GitHub Pull Requests extension](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github). You can assign any issue or PR to Copilot Coding Agent right from the sidebar - no need to switch to your browser. Just select, assign, and let the agent take it from there.
+
+![A GIF showing Copilot assigned to an issue from within VS Code](assign-to-copilot-gif.gif)
+
+### Track your agent’s progress
+
+There’s a new **Copilot on My Behalf** query in the Pull Requests view. This shows you everything Copilot is working on for you. Want to see exactly what the agent did? Click "View Session" and watch the play-by-play. You get full transparency into every command and decision. You can also terminate the agent - all without leaving the editor.
+
+![A WIP PR shown under the "Copilot on my behalf" section with the "View Session" button highlighted](copilot-wip.png)
+
+In this screenshot, the agent implemented a trending section complete with UI and database changes. All of this is one interaction so it's only one premium request.
+
+### Review, comment, and iterate
+
+When the agent is done, it will request your review on it's PR and assign the PR to you. It even includes a screenshot if applicable so you can quickly verify whether it did the right thing or not at all.
+
+![PR from Coding agent shown in VS Code with a screenshot](draft-with-screenshot.png)
+
+You can review the agent’s PR, leave comments, and ask for changes - all from within VS Code. The agent will picks up your feedback, updates the PR and let you know when it's finished.
+
+My colleague @pierceboggan pointed out a fascinating workflow here - with services like Azure Static Web Apps, Vercel and Netlify that offer previews for PR's, you can review the agent's work without even having to checkout it's PR and run it locally.
+
+![Live preview of the site on Netlify but we have an error](live-preview-netlify.png)
+
+Uh oh - looks like we didn't quite get the trending working. In this case, that's ok. In fact it's good. We didn't get it because our agent doesn't have access to create the required database changes so it created a migraiton script for us instead as part of the PR.
+
+But what if we wanted to give it access to make the database changes to a dev environment? Coding agent has support for MCP servers and provided you have one for your database of choice (in this case supabse), you can [configure MCP servers for the agent on the repo](https://docs.github.com/en/copilot/how-tos/agents/copilot-coding-agent/extending-copilot-coding-agent-with-mcp). It's up to you - how autonomous do you want the agent to be?
+
+![MCP configuration shown for copilot coding agent on github.com](coding-agent-mcp-config.png)
+
+But it gets even better - because you can do all of this right from the agent in VS Code.
```

Comment:
```suggestion
But it gets even better - because you can do all of this right from VS Code.
```

This is iffy because the screenshot is GitHub - so it looks like you can't configure MCP for coding agent from within VS Code. So even if you meant it as a transition into the next section, I think it feels a little confusing and a little repetitive (because "within VS Code", "directly in VS Code", "right from the chat in VS Code" is used a few times). You might be able to drop this line altogether.